tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_advertiser.c

```diff
@@ -262,6 +386,86 @@ static void test_tx_cb_multi(void)
 	PASS();
 }
 
+static void test_tx_proxy_mixin(void)
+{
+	static struct bt_mesh_prov prov = {
+		.uuid = test_prov_uuid,
+	};
+	int err;
+
+	/* Initialize mesh stack and enable pb gatt bearer to emit beacons. */
+	bt_mesh_device_setup(&prov, &comp);
+	err = bt_mesh_prov_enable(BT_MESH_PROV_GATT);
+	ASSERT_OK(err, "Failed to enable GATT provisioner");
+
+	/* start pb gatt beacons with 100 msec interval */
+	k_sleep(K_MSEC(1800));
+
+	LOG_INF("Provision device under test");
+	/* Provision dut and start gatt proxy beacons. */
+	bt_mesh_provision(test_net_key, 0, 0, 0, adv_cfg.addr, adv_cfg.dev_key);
+
+	/* start proxy beacons with 1 sec interval */
+	k_sleep(K_MSEC(6000));
+
+	/* send adv data */
+	struct net_buf *buf = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_LOCAL_ADV,
+			BT_MESH_TRANSMIT(5, 20), K_NO_WAIT);
+	net_buf_add_mem(buf, txt_msg, sizeof(txt_msg));
+	bt_mesh_adv_send(buf, NULL, NULL);
+	k_sleep(K_MSEC(150));
+
+	/* start proxy beacons again */
+	k_sleep(K_MSEC(6000));
```

Comment:
I guess the sleep sequences here for the most part is put in to sync with the RX counterpart? Should we consider merging the sync lib from mesh-1.1 branch to zephyr? IMO using ad-hoc sleep cycles to make the test work is error proned. Im not necessarily saying we must do this before merging this, but it should be considered.

Reply:
The test is synced in a couple of places automatically using K_FOREVER semaphores. If rx part has been completed something faster it will wait for tx part in semaphore implicitly. I guess it removes potential instability.

Reply:
btw. the test works for both extended advertiser and legacy advertiser for different PCs. Obviously, advertisers have different timelines.